### DISCUSSION TITLE
Rename 'images' in Seurat object

### DISCUSSION THREAD

**Discussion by cathalgking** [satijalab/seurat#6953] [Q&A]
Hi. I have 4 images in my Seurat object that were read in via the read10x() function individually and then merged. The images came from 1 slide of a 10x Visium experiment (1 from each of the 4 capture areas). I would like to rename the image names. Now they are called ''slice1'', slice.1.1'' etc. How can I do this?

![Screen Shot 2023-02-15 at 2 33 17 pm](https://user-images.githubusercontent.com/32261323/218926850-61bf217b-22f6-4f50-8a18-27055f637458.png)

Thanks

**Comment by Jiayi-Zheng** [comment]
I've never tried visium with seurat but based on the formatting I saw, try:
`seu_spe$images$YOURNAME = seu_spe$images$slice1`
`seu_spe$images$slice1 = NULL`?

**Comment by mcrewcow** [comment]
Hi cathalgking,

You could try copying the image (seurat_object@images$new_name <- ...$old_name) yet the reason you have those labels is because of the key parameter.
Next time you create the objects, set the names in the keys (key = "image_") to get the proper annotation.

**Reply by fangyang-cbio** [reply]
Would you explain how to set the keys name? Is it in Load10X_Spatial()? Thank you!

**Comment by RheinerAlkohol** [comment]
Hey,

I used Load10X_Spatial("yourdir", slice = "image_name").

**Comment by keenhl** [comment]
You could do something like this.

names(object@images) <- image_names
